2025/incoming/stopping-the-evil-grinch-a-holiday-defense-guide.pod

```diff
@@ -0,0 +1,311 @@
+
+=encoding utf8
+
+=for :html
+<img src="elf_runnig_perl_script.png" alt="Stopping the Evil Grinch" style="float: right; margin: 0 0 1em 1em; width: 300px;"/>
+
+During a December evening in Santa's workshop, the security team received an urgent alert:
+a malicious actor, known as the "Evil Grinch," intended to compromise the systems running
+the toy production environment. Fortunately, the team already relied on several essential
+security tools:
+
+=over 4
+
+=item * Lynis for system auditing
+
+=item * ClamAV for malware scanning
+
+=item * Perl to orchestrate everything
+
+=back
+
+=head2 Installing Lynis Without Sudo
+
+Because the environment restricted sudo usage, Lynis was
+installed in the user's home directory using the following commands:
+
+  #!vim bash
+  wget https://downloads.cisofy.com/lynis/lynis-3.1.6.tar.gz
+  tar -xvf lynis-3.1.6.tar.gz
+  mkdir -p /home/user/bin
+  chmod 775 /home/user/bin
+
+With this wrapper script, Lynis was executable without requiring sudo privileges.
+
+  cat << 'EOF' > /home/user/bin/lynis
+  #!/bin/bash
+  LYNIS_DIR="/home/user/lynis-3.1.6"
+  export LYNIS_LOG_FILE="/home/user/lynis-logs/lynis.log"
+  export LYNIS_REPORT_FILE="/home/user/lynis-logs/lynis-report.dat"
+  cd "$LYNIS_DIR" || exit 1
+  ./lynis "$@"
+  EOF
+
+  chmod 775 /home/user/bin/lynis
+
+=head2 Installing ClamAV
+
+ClamAV was installed using the system package manager:
+
+  #!vim bash
+  sudo apt update
+  sudo apt install clamav clamav-daemon -y
+
+=head2 Building the Perl Cron Script
+
+A Perl script was created to automate security reporting and deliver the results via email.
+
+First, the necessary Perl dependencies were installed:
+
+  #!vim bash
+  cpanm Moo \
+        Email::Sender::Simple \
+        Email::Sender::Transport::SMTP \
+        Email::MIME \
+        Try::Tiny \
+        Types::Standard \
+        IO::All \
+        DateTime \
+        Readonly \
+        Log::Log4perl
+
+The script is meant to run daily and exits early if it is not the last day of the month,
+determined via:
+
+  DateTime->now()->is_last_day_of_month()
+
+Logging is handled by the Log::Log4perl module. The core of the script is the run_report
+function, which deletes any previous report file and executes each command.
+Because ClamAV errors are expected when scanning protected or locked files,
+the ClamAV call is allowed to return a non-zero exit code.
+
+Below is the Perl script:
+
+  #!vim perl
+  #!/usr/bin/env perl
+
+  use strict;
+  use warnings;
+
+  use DateTime;
+  use Cwd;
+  use Readonly;
+  use Log::Log4perl qw(:easy);
+  use Dotenv;
+
+  Log::Log4perl->easy_init(
+      {
+          level => $DEBUG,
+          file  => ">>test.log"
+      }
+  );
+
+  Readonly::Scalar my $lynis_file  => '/home/user/lynis-report.dat';
+  Readonly::Scalar my $clamav_file => '/home/user/clamav.log';
+
+  if ( DateTime->now()->is_last_day_of_month() ) {
+
+      $logger->debug('Last day of month detected. Preparing reports.');
```

Comment:
`$logger` has not been initialized. Can you run the code snippets to ensure that they compile?

Reply:
fixed